### DISCUSSION TITLE
Documentation Topics

### DISCUSSION THREAD

**Discussion by karimStekelenburg** [animo/aries-javascript-docs#65] [Ideas]
The purpose of this discussion is to gather and create a list of topics that we think need to be documented. The end goal is to have an overview/checklist of topics/features that require documentation.

If applicable, please add links to issues, PRs or discussions of other Aries JS related repositories.

**Comment by 2byrds** [comment]
Awesome!  Lets target this discussion at the 20220804 Aries JavaScript meeting. @NB-MikeRichardson we'd love to see your bulleted list.  I'll fish for documentation related issues in the other repos.

**Comment by 2byrds** [comment]
In general I think developers prefer templates, CI, code and other automated checks over reading through documentation.  For instance, good discussion here about templated checklist for PRs/Issues https://github.com/hyperledger/aries-framework-javascript/discussions/914

**Reply by 2byrds** [reply]
From our meeting today, issue templates are very useful.  PR template should be more general, just list what you did... a very minimal template. We'd like CI will catch most things.

**Comment by 2byrds** [comment]
This is a good example of a documentation request in AFJ, that was then also ported into Javascript Docs:
https://github.com/hyperledger/aries-framework-javascript/issues/771
https://github.com/hyperledger/aries-framework-javascript/pull/824
and related JS Docs:
https://github.com/hyperledger/aries-javascript-docs/tree/548aea1e82734c80e04f2efdf07fbc3ed310f1db/guides/tutorials/postgres-database-nodejs

**Comment by 2byrds** [comment]
From our meeting today, primarily we want to bootstrap new developers to speed momentum into AFJ:

- Tricks like deleting the wallet if things get messed up during development
- Running the tests
- Can we share developer notes and promote especially helpful diagrams, information, etc? Perhaps the dev fork can be linked to?
- How to add a module?
- Programmatically create class diagrams?

**Comment by TimoGlastra** [comment]
I recently created a template for requesting features of AFJ to be documented, might be useful for this: https://github.com/hyperledger/aries-javascript-docs/issues/new?assignees=&labels=request&template=documentation-request.yml&title=%5BRequest%5D%3A+

**Comment by NB-MikeRichardson** [comment]
Sure. I’ll see if I can flesh out my bullet list a bit more and maybe we can discuss these.
